### ISSUE TITLE
CD Release Apple App Store Promote - new version

### ISSUE THREAD

**Issue by timClyburn** [Microsoft/app-store-vsts-extension#93]
I have a CD process to release my new versions of an app to TestFlight and then, once tested and approved, promoted to the app store for review. If we forget to create a new version in the app store prior to running this promotion then it fails as it's unable to find the version in the store. I know this is expected behaviour.

So my question is, is there a way to automatically create this new version in the app store during the promotion step? Anything that reduces/removes manual intervention is a benefit.

Thanks

**Comment by tipa** [comment]
Where is it stated that it is expected behavior for the task to fail when not manually creating a new version in the iTunes portal? Have you found any solution or workaround for this?

**Comment by timClyburn** [comment]
> Where is it stated that it is expected behavior for the task to fail when not manually creating a new version in the iTunes portal? Have you found any solution or workaround for this?

I don't think that it is stated explicitly, however trying to promote a build into release failed if that build number has not been created within app store connect.
